src/plugins/data/public/search/expressions/esaggs/build_tabular_inspector_data.ts

```diff
@@ -18,35 +18,41 @@
  */
 
 import { set } from '@elastic/safer-lodash-set';
-import { FormattedData } from '../../../../../plugins/inspector/public';
-import { TabbedTable } from '../../../common';
-import { FormatFactory } from '../../../common/field_formats/utils';
-import { createFilter } from './create_filter';
+import {
+  FormattedData,
+  TabularData,
+  TabularDataValue,
+} from '../../../../../../plugins/inspector/common';
+import { Filter, TabbedTable } from '../../../../common';
+import { FormatFactory } from '../../../../common/field_formats/utils';
+import { createFilter } from '../create_filter';
 
 /**
- * @deprecated
+ * Type borrowed from the client-side FilterManager['addFilters'].
  *
- * Do not use this function.
- *
- * @todo This function is used only by Courier. Courier will
- *   soon be removed, and this function will be deleted, too. If Courier is not removed,
- *   move this function inside Courier.
- *
- * ---
+ * We need to use a custom type to make this isomorphic since FilterManager
+ * doesn't exist on the server.
  *
+ * @internal
+ */
+export type AddFilters = (filters: Filter[] | Filter, pinFilterStatus?: boolean) => void;
+
+/**
  * This function builds tabular data from the response and attaches it to the
  * inspector. It will only be called when the data view in the inspector is opened.
+ *
```

Comment:
I think we can get rid of this function (maybe in a separate PR). We had this to support inspector in expressions in the early days but i think right now its no longer needed. Rather, we should:
- log the datatable in every chart function (as esaggs is not the right place, futrther modifications to the datatable could happen further in expression)
- update inspector view to work with new datatable format